plugins/woocommerce/client/admin/client/settings-payments/components/buttons/enable-gateway-button.tsx

```diff
@@ -124,6 +131,12 @@ export const EnableGatewayButton = ( {
 				if ( response.data === 'needs_setup' ) {
 					// We only need to perform additional logic/redirects if no account connected.
 					if ( ! gatewayState.account_connected ) {
+						// Record the event when user successfully enables a gateway.
+						recordEvent( 'settings_payments_provider_enable', {
```

Comment:
I am a bit shaky if it's the right place to capture this event.

Reply:
We'll leave it like this for now as moving this event to backend PHP code will require hunting down the `woocommerce_toggle_gateway_enabled` action, which may not be worth the effort. Slack discussion: p1738751326045739-slack-C0800LPNCET

Reply:
On the latest update, this seems to work nicely so I'm happy to leave it like this.